### DISCUSSION TITLE
TheItemContainerTheme inheritance Behavior

### DISCUSSION THREAD

**Discussion by workgroupengineering** [AvaloniaUI/Avalonia#10060] [Q&A] [answered]
In discussion  #10018 is answer the Equivalent  of  ItemsControl.ItemContainerStyle is Menu.ItemContainerTheme but the behavior isn't the same.

Considering this WPF example:

```xaml
        <Menu ItemsSource="{Binding Columns}">
            <Menu.ItemContainerStyle>
                <Style TargetType="MenuItem">
                    <Setter Property="Header" Value="{Binding Description}"/>
                    <Setter Property="ItemsSource" Value="{Binding Columns}"/>
                </Style>
            </Menu.ItemContainerStyle>
        </Menu>
```

will produce a result like this:

![WPF behavior](https://user-images.githubusercontent.com/12531229/214051907-e19bda43-3d89-4f5d-b38d-ab41452771ef.png)

The equivalent in Avalonia would be:

```xaml
    <Menu Items="{Binding Columns}">
      <Menu.ItemContainerTheme>
        <ControlTheme TargetType="MenuItem" x:DataType="l:Column" BasedOn="{StaticResource {x:Type MenuItem}}">
          <Setter Property="Header" Value="{Binding Description}"/>
          <Setter Property="Items" Value="{Binding Columns}"/>
        </ControlTheme>
      </Menu.ItemContainerTheme>
    </Menu>
```

will produce a result like this:

![Avalonia behavior](https://user-images.githubusercontent.com/12531229/214052963-01bd1766-e0a9-4128-9061-70de2e9b1548.png)

As you can see, other than incorrect styling, submenus do not inherit the ItemContainerTheme.
is it a desired behavior? If so, how can I emulate WPF behavior?

**Comment by maxkatz6** [comment, marked as answer]
In general it can be done with styles, see https://github.com/AvaloniaUI/Documentation/issues/381#issue-1485800924 https://github.com/AvaloniaUI/Avalonia/issues/3100#issuecomment-753452752

But this difference with WPF indeed looks strange.

**Reply by workgroupengineering** [reply]
Thanks for the reply. For the moment I use this solution.
Can I do a PR that respects WPF behavior?

**Reply by maxkatz6** [reply]
Can you create an issue from this so it can be discussed there?
Also, we have now that in Avalonia Menu.ItemContainerTheme changes theme of top-level menu items. While in WPF it changes sub-menu items.
In WPF as well initial style of these menu items is different, how do developers change theme of the top-level items specifically?

**Reply by workgroupengineering** [reply]
Add PseudoClass `:IsTop` and using Selector inside ControlTheme

**Reply by maxkatz6** [reply]
No, my question was about wpf. I don't see how you can style them differently in wpf.

**Reply by workgroupengineering** [reply]
You must create ControlTemplate  like this:
```xaml
<ControlTemplate x:Key="{x:Static MenuItem.TopLevelHeaderTemplateKey}"
                 TargetType="{x:Type MenuItem}">
  ...
</ControlTemplate/>
```

You can find other info [here](https://learn.microsoft.com/en-us/dotnet/desktop/wpf/controls/menu-styles-and-templates?view=netframeworkdesktop-4.8).
